tests/sentry/models/test_team.py

```diff
@@ -150,3 +159,49 @@ def test_release_project_envs(self):
         assert not ReleaseProjectEnvironment.objects.filter(
             release=release, project=project, environment=env
         ).exists()
+
+
+@region_silo_test
+class TeamDeletionTest(TestCase):
+    def test_cannot_delete_with_queryset(self):
+        team = self.create_team(self.organization)
+        assert Team.objects.filter(id=team.id).exists()
+        with pytest.raises(ProgrammingError), transaction.atomic():
+            Team.objects.filter(id=team.id).delete()
+        assert Team.objects.filter(id=team.id).exists()
+
+    def test_hybrid_cloud_deletion(self):
+        team = self.create_team(self.organization)
+        NotificationSetting.objects.update_settings(
+            ExternalProviders.EMAIL,
+            NotificationSettingTypes.ISSUE_ALERTS,
+            NotificationSettingOptionValues.ALWAYS,
+            team=team,
+        )
+
+        assert Team.objects.filter(id=team.id).exists()
+        assert NotificationSetting.objects.find_settings(
+            provider=ExternalProviders.EMAIL,
+            type=NotificationSettingTypes.ISSUE_ALERTS,
+            team=team,
+        ).exists()
+
+        with outbox_runner():
+            team.delete()
+
+        assert not Team.objects.filter(id=team.id).exists()
+        assert NotificationSetting.objects.find_settings(
+            provider=ExternalProviders.EMAIL,
+            type=NotificationSettingTypes.ISSUE_ALERTS,
+            team=team,
+        ).exists()
```

Comment:
The test fails here because the NotificationSetting is cascade deleted with the team.
Any hints?
I commented out this line and verified that the call to `schedule_hybrid_cloud_foreign_key_jobs` below results in a call to `process_hybrid_cloud_foreign_key_cascade_batch` for `NotificationSetting.team_id`.

Reply:
i'm confused, don't you _want_ this to be deleted too?

Reply:
Yes, it is deleted via Django cascade. I was wondering how to setup a test case to verify that the delete would be handled by `schedule_hybrid_cloud_foreign_key_jobs` in a control/region silo setup.
Perhaps this test is sufficient if I remove this line.

Reply:
mark the test decorator with `stable=True` argument and the `outbox_runner` will take care of it.

Reply:
Does `stable=True` require a service for notification settings?
`notification_setting_service` is in progress. Merging this PR first.